src/lua/httpc.c

```diff
@@ -155,15 +155,36 @@ luaT_httpc_request(lua_State *L)
 	if (req == NULL)
 		return luaT_error(L);
 
+	const char* content_type = NULL;
 	double timeout = TIMEOUT_INFINITY;
 	int max_header_name_length = MAX_HTTP_HEADER_NAME_LEN;
 	if (lua_isstring(L, 4)) {
 		size_t len = 0;
 		const char *body = lua_tolstring(L, 4, &len);
+		if (len > 0 && httpc_set_body(req, body, len) != 0) {
+			httpc_request_delete(req);
+			return luaT_error(L);
+                }
+        } else if ((lua_istable(L, 4)) &&
+                   (strcasecmp(method, "post") == 0)) {
+		/* serialize body to JSON */
+		lua_getglobal(L, "require");
+		lua_pushstring(L, "json");
+		if (luaT_call(L, 1, 1) != 0)
+			return luaT_error(L);
+		lua_getfield(L, -1, "encode");
+		lua_pushvalue(L, 4);
+		if (luaT_call(L, 1, 1) != 0)
+			return luaT_error(L);
```

Comment:
Nit: We can safe the reference to the function in the Lua registry at initialization and access it here faster. Also we can avoid `require` and access the module as `package.loaded.json` (however I doubt that there is any real difference; just another variant of the same).